src/Datadog.Trace/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+
+namespace Datadog.Trace.Plugins
+{
+    internal static class PluginManager
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(PluginManager));
+
+        public static IReadOnlyCollection<IOTelPlugin> Loaded { get; private set; } = ArrayHelper.Empty<IOTelPlugin>();
+
+        internal static void TryLoadPlugins(JsonConfigurationSource pluginsConfig)
+        {
+            if (pluginsConfig == null)
+            {
+                return;
+            }
+
+            var runtimeName = GetPluginsRuntimeName();
+
+            Log.Debug("Executing plugins configuration: {0}", pluginsConfig);
+            Log.Information("Trying to load plugins for '{0}' runtime.", runtimeName);
+
+            // TODO: Detect if objects instead of path
+            var pluginFiles = pluginsConfig?.GetValue<JToken>($"['{runtimeName}']").ToObject<string[]>();
```

Comment:
Optional security options could be added. Currently parses only path, in future could be object with path and checksum etc.

Reply:
As far as I see, currently, if user would provide improper JSON then it will silently fail, because of:

```
            try
            {
                // Creates GlobalSettings instance and loads plugins
                PluginManager.TryLoadPlugins(GlobalSettings.Source.PluginsConfiguration);

                // First call to create Tracer instace
                _ = Tracer.Instance;
            }
            catch
            {
                // ignore
            }
```

Also, the tracer may even not be initialized or initialize at some strange time?

Reply:
Actually not because of the try catch there but because of the existing logic moved to `TryLoadJsonConfigurationFile`.